release function-mesh 0.14.0 (#653)
streamnative/function-mesh: CHANGELOG.md

# Function Mesh v0.14.0 Release Notes

## v0.14.0 What's New

* [Runner Image] Support Pulsar 3.0 ([#625](https://github.com/streamnative/function-mesh/pull/625))

> * [Helm Charts] Allows the `function-mesh-secrets-webhook` helm chart to be deployed into non-default kubernetes namespaces ([#628](https://github.com/streamnative/function-mesh/pull/628))

Reviewer: ```suggestion
* [Helm Charts] Allow the `function-mesh-secrets-webhook` Helm chart to be deployed into non-default Kubernetes namespaces ([#628](https://github.com/streamnative/function-mesh/pull/628))
```

> * [Controller] Support `CleanupSubscription` ([#622](https://github.com/streamnative/function-mesh/pull/622))

Reviewer: ```suggestion
* [Controller] Support the `CleanupSubscription` configuration ([#622](https://github.com/streamnative/function-mesh/pull/622))
```

> * [CI] Trigger release action when tag with `v` prefix is pushed ([#631](https://github.com/streamnative/function-mesh/pull/631))

Reviewer: ```suggestion
* [CI] Trigger the release action when the tag with the prefix `v` is pushed ([#631](https://github.com/streamnative/function-mesh/pull/631))
```

> * [CRD] Support `PersistentVolumeClaimRetentionPolicy` ([#633](https://github.com/streamnative/function-mesh/pull/633))

Reviewer: ```suggestion
* [CRD] Support the `PersistentVolumeClaimRetentionPolicy` configuration ([#633](https://github.com/streamnative/function-mesh/pull/633))
```

* [Runner Image] Release runner images with `pulsarctl` ([#630](https://github.com/streamnative/function-mesh/pull/630))
* [Controller] Sync to the latest Pulsar Function Proto files ([#637](https://github.com/streamnative/function-mesh/pull/637))
* [Controller] Upgrade to Golang 1.20.4 ([#641](https://github.com/streamnative/function-mesh/pull/641))

> * [Controller] Prevent cleanup resources if managed by the function-mesh-worker-service ([#640](https://github.com/streamnative/function-mesh/pull/640))

Reviewer: ```suggestion
* [Controller] Prevent cleaning up resources that are managed by the Function Mesh Worker service ([#640](https://github.com/streamnative/function-mesh/pull/640))
```
